controller/internal/enforcer/acls/acl.go

```diff
@@ -42,6 +42,13 @@ func (a *acl) addToCache(ip net.IP, mask int, port string, policy *policy.FlowPo
 		portList = val.(portActionList)
 	}
 
+	/* check if this is duplicate entry */
+	for _, portAction := range portList {
+		if *r == *portAction {
+			return nil
+		}
+	}
+
```

Comment:
Any chance of nils here?

Reply:
No. We never store nil pointers in the portList array.